include/micm/solver/linear_solver.hpp

```diff
@@ -57,6 +57,11 @@ namespace micm
     /// @brief default constructor
     LinearSolver(){};
 
+    LinearSolver(const LinearSolver&) = delete;
+    LinearSolver& operator=(const LinearSolver&) = delete;
```

Comment:
what is the reason to delete the copy constructor and assignment operator for almost all the class in this PR?

Reply:
It was the combination of a few things. First, we don't actually need the copy constructor/assignment for anything we're currently doing. Second, for the JIT classes, we would have to figure out how to copy the generated JIT functions and resource managers, and it wasn't obvious how to do this. And lastly, for the remaining classes, I removed them so that the solver builder follows the same logic for all solvers (JIT and non-JIT).

Reply:
Thanks for the clarification.